client/src/app/hooks/useFetchArchetypeOrApplicationByAssessment.ts

```diff
@@ -0,0 +1,42 @@
+import { useQuery } from "@tanstack/react-query";
+import { getApplicationById, getArchetypeById } from "@app/api/rest";
+
+const ArchetypeOrApplicationByAssessmentQueryKey =
+  "archetypeOrApplicationByAssessment";
+
+export function useFetchArchetypeOrApplicationByAssessment(
+  isArchetype?: boolean,
+  itemId?: number | string
+) {
+  const { data, isLoading, isError, error } = useQuery(
+    [
+      ArchetypeOrApplicationByAssessmentQueryKey,
+      isArchetype ? "archetype" : "application",
+      itemId,
+    ],
+    async () => {
+      if (!itemId) {
+        return null;
+      }
+
+      if (isArchetype) {
+        const archetypeData = await getArchetypeById(itemId);
+        return archetypeData;
+      } else {
+        const applicationData = await getApplicationById(itemId);
+        return applicationData;
+      }
+    },
+    {
+      enabled: !!itemId,
+    }
+  );
+
+  return {
+    data,
+    isLoading,
+    isError,
+    error,
+    isArchetype,
+  };
+}
```

Comment:
So, IMO this coupling of the two types of data being fetched into one useQuery isn't necessary. We could keep the two hooks separate and call both, but just pass `enabled: false` to the one that isn't relevant (expose that as an argument on the hook).

But I understand that would mean undoing a chunk of your work so I'm fine with leaving it as-is in the interest of time if necessary.

Reply:
Not a big deal to change. I think that is probably a good call.